tempesta_fw/http.c

```diff
@@ -2241,6 +2241,8 @@ __tfw_http_resp_fwd(TfwCliConn *cli_conn, struct list_head *ret_queue)
 		BUG_ON(!req->resp);
 		tfw_http_resp_init_ss_flags(req->resp, req);
 		if (tfw_cli_conn_send(cli_conn, (TfwMsg *)req->resp)) {
+			list_del_init(&req->msg.seq_list);
+			tfw_http_resp_pair_free(req);
```

Comment:
Correct, `tfw_http_resp_fwd()` takes care on unforwarded responses, and all the responses are destroyed there. The new line `list_del_init(&req->msg.seq_list);` was missed there thought, so #1099 happen. This exact lines are not required here since we still do the same later during error handling at the end of `tfw_http_resp_fwd()` .

Reply:
> why do we need to free requests and responses here? Don't we need to adjust or completely remove the loop at the end of tfw_http_resp_fwd()?

It's possible to move all the lines from the loop to `__tfw_http_resp_fwd()`, but there will be no simplification other than moving all reclamation code together. Now two sites are apart from each another. That's slightly confusing. However, that much is fine.